src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -387,6 +387,10 @@ func chained(rootCA string, persistent bool) (http.RoundTripper, error) {
 			ClientSessionCache: clientSessionCache,
 		},
 	}
+	if persistent {
```

Comment:
Oh, I see what you're saying.  DisableKeepAlives doesn't just add a Connection: close header, it actually closes the connection itself even if the server didn't.  See line 513 and 524 of [transport.go](https://golang.org/src/net/http/transport.go).